server/src/metrics/edge_metrics.rs

```diff
@@ -76,6 +85,25 @@ impl EdgeInstanceData {
             latency_upstream: UpstreamLatency::default(),
             connected_edges: vec![],
             connected_streaming_clients: 0,
+            requests_since_last_report: DashMap::default(),
+        }
+    }
+
+    pub fn observe_request(&self, http_target: &str, status_code: i64) {
+        match status_code {
+            200 => {
+                self.requests_since_last_report
+                    .entry(http_target.to_string())
+                    .or_default()
+                    .requests_200 += 1
```

Comment:
I think this isn't thread safe. Docs are a bit hard to grok here but dashmap controls request_200 but I don't think it's able to wrap access and modify in in the same lock across function calls

Can't prove it but this'd keep me up at night